pkg/errors/errors.go

```diff
@@ -233,7 +233,27 @@ func NewUserNotAllowedError(image string, onbuild bool) error {
 	return Error{
 		Message:    msg,
 		ErrorCode:  UserNotAllowedError,
-		Suggestion: fmt.Sprintf("modify image %q to use a numeric user within the allowed range or build without the --allowed-uids flag", image),
+		Suggestion: fmt.Sprintf("modify image %q to use a numeric user within the allowed range, or build without the --allowed-uids flag", image),
+	}
+}
+
+// NewAssembleUserNotAllowedError returns a new error that indicates that the build
+// could not run because the build or image uses an assemble user outside of the range
+// of allowed users.
+func NewAssembleUserNotAllowedError(image string, assembleUserOverride string) error {
+	var msg, suggestion string
+	if len(assembleUserOverride) > 0 {
+		msg = fmt.Sprintf("assemble user %q must be numeric and within the range of allowed users", assembleUserOverride)
+		suggestion = "build without the --allowed-uids flag or the --assemble-user flag"
+	} else {
+		assembleLabel := "io.openshift.s2i.assemble-user"
```

Comment:
I wish I could use the `AssembleUserLabel` constant, but `errors` is used by every other package in s2i.

Reply:
meaning you're getting a circular dependency when you reference the constant?

Reply:
correct. I'm reluctant to create a standalone `constants` package just for this purpose.

Reply:
yeah that's fine